compiler-plugin/src/main/kotlin/arrow/meta/quotes/ForExpression.kt

```diff
@@ -0,0 +1,36 @@
+package arrow.meta.quotes
+
+import arrow.meta.Meta
+import arrow.meta.phases.ExtensionPhase
+import org.jetbrains.kotlin.psi.KtDestructuringDeclaration
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtForExpression
+
+/**
+ * A [KtForExpression] [Quote] with a custom template destructuring [ForExpressionScope]
+ *
+ * @param match designed to to feed in any kind of [KtForExpression] predicate returning a [Boolean]
+ * @param map map a function that maps over the resulting action from matching on the transformation at the PSI level.
+ *
+ * forExpression({ true }) { e ->
+ *   Transform.replace(
+ *    replacing = e,
+ *    newDeclaration = """ for $`(param)` { $body } """.`for`
+ *   )
+ *  }
```

Comment:
Here' s a cool thing - we can add ``` here like:

```
 *``````kotlin:ank:silent
 * forExpression({ true }) { e ->
 *   Transform.replace(
 *    replacing = e,
 *    newDeclaration = """ for $`(param)` { $body } """.`for`
 *   )
 *  }
 * ```
```

Reply:
Awesome job!